### DISCUSSION TITLE
Is it possible to convert a normal Pytorch checkpoint to a deepspeed-compatible checkpoint?

### DISCUSSION THREAD

**Discussion by richarddwang** [deepspeedai/DeepSpeed#3027] [Q&A] [answered]
My goal is to use deepspeed zero inference on bloomz model, or even any model on huggingface hub. 

Specifically, I would like to download model weights from huggingface hub repo (e.g. [bloomz](https://huggingface.co/bigscience/bloomz/tree/main)), and convert them to deepspeed compatible checkpoints through a script. (basically a reverse one of [zero_to_fp32.py](https://github.com/microsoft/DeepSpeed/blob/master/deepspeed/utils/zero_to_fp32.py)?)

Are there existing scripts for this or similar purpose?
Or is it possible to write one? (then where should I start?)

**Comment by tjruwase** [comment, marked as answer]
This should work out of the box, please see https://github.com/huggingface/transformers-bloom-inference/blob/main/bloom-inference-scripts/bloom-ds-zero-inference.py

**Reply by richarddwang** [reply]
Didn't notice the script is also compatible to zero inference. Thank you! You saved me a lot of time.
